### ISSUE TITLE
gf should expand environment variables in filepaths.

### ISSUE THREAD

**Issue by amal-khailtash** [VSCodeVim/Vim#2680]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
No

**What happened**:
Nothing :)

**What did you expect to happen**:
I expected to go to a file that includes environment variable names.

**How to reproduce it (as minimally and precisely as possible)**:
Create a text file with the following lines:
```
$HOME/.bashrc
${HOME}/.bashrc
```
Put cursor on one of those files and press gf.
If cursor is on the first line, vscode does not do anything.
If cursor is one ${HOME}, vscode would create a new file called HOME.

**Environment**:
RHEL 7.2

- Extension (VsCodeVim) version:
- VSCode version: 1.23.1
- OS version: 0.12.0

**Comment by jpoon** [comment]
Is this something that is even supported with vanilla vim? The only thing I've come across is http://vim.wikia.com/wiki/Open_file_under_cursor "Using an environment variable" section.

We would need to call out to the shell to ascertain if `${ENV}` meant anything, and if it does retrieve the value (in a cross-platform manner). I fear perf would take a hit as we'd need to (1) spawn a separate process and (2) there's no way to preload/cache any of this data (as environment variables can change outside vscode) so we'd need to do this on every file load operation that has a `${ENV}` pattern.

**Comment by amal-khailtash** [comment]
Yes, this is supported by vanilla vim.  The first one "$HOME/.bashrc" is supported without any user settings, to get the second one "${HOME}/.bashrc" working, one needs to add { and } to the list of characters that is recognized as file names:

```
" Expand variables before gf, gF
set isfname+={,}
```
http://vimdoc.sourceforge.net/htmldoc/editing.html#gf
http://vimdoc.sourceforge.net/htmldoc/options.html#'isfname'

**Comment by robjac** [comment]
gf no longer works for me. -- It is intermittent at best, but I've noticed that it definitely always fails on files in the same folder the file the cursor is on.
